src/main/scala/net/psforever/actors/session/support/SessionData.scala

```diff
@@ -1640,92 +1403,392 @@ class SessionData(
+    progressBarUpdate.cancel()
+    progressBarValue = None
+  }
+
+  def handleTrade(pkt: TradeMessage): Unit = {
+    val TradeMessage(trade) = pkt
+    log.trace(s"${player.Name} wants to trade for some reason - $trade")
+  }
+
+  def handleDisplayedAward(pkt: DisplayedAwardMessage): Unit = {
+    val DisplayedAwardMessage(_, ribbon, bar) = pkt
+    log.trace(s"${player.Name} changed the $bar displayed award ribbon to $ribbon")
+    avatarActor ! AvatarActor.SetRibbon(ribbon, bar)
+  }
+
+  def handleObjectDetected(pkt: ObjectDetectedMessage): Unit = {
+    val ObjectDetectedMessage(_, _, _, targets) = pkt
+    shooting.FindWeapon.foreach {
+      case weapon if weapon.Projectile.AutoLock =>
+        //projectile with auto-lock instigates a warning on the target
+        val detectedTargets = shooting.FindDetectedProjectileTargets(targets)
+        val mode = 7 + (weapon.Projectile == GlobalDefinitions.wasp_rocket_projectile)
+        detectedTargets.foreach { target =>
+          continent.AvatarEvents ! AvatarServiceMessage(target, AvatarAction.ProjectileAutoLockAwareness(mode))
+        }
+      case _ => ()
+    }
+  }
+
+  def handleTargetingImplantRequest(pkt: TargetingImplantRequest): Unit = {
+    val TargetingImplantRequest(list) = pkt
+    val targetInfo: List[TargetInfo] = list.flatMap { x =>
+      continent.GUID(x.target_guid) match {
+        case Some(player: Player) =>
+          val health = player.Health.toFloat / player.MaxHealth
+          val armor = if (player.MaxArmor > 0) {
+            player.Armor.toFloat / player.MaxArmor
+          } else {
+            0
+          }
+          Some(TargetInfo(player.GUID, health, armor))
+        case _ =>
+          log.warn(
+            s"TargetingImplantRequest: the info that ${player.Name} requested for target ${x.target_guid} is not for a player"
+          )
+          None
+      }
+    }
+    sendResponse(TargetingInfoMessage(targetInfo))
+  }
+
+  def handleHitHint(pkt: HitHint): Unit = {
+    val HitHint(_, _) = pkt
+  }
+
+  /* messages */
+
+  def handleSetAvatar(avatar: Avatar): Unit = {
+    session = session.copy(avatar = avatar)
+    if (session.player != null) {
+      session.player.avatar = avatar
+    }
+    LivePlayerList.Update(avatar.id, avatar)
+  }
+
+  def handleReceiveAccountData(account: Account): Unit = {
+    log.trace(s"ReceiveAccountData $account")
+    session = session.copy(account = account)
+    avatarActor ! AvatarActor.SetAccount(account)
+  }
+
+  def handleUpdateIgnoredPlayers(pkt: PlanetSideGamePacket): Unit = {
+    pkt match {
+      case msg: FriendsResponse =>
+        sendResponse(msg)
+        msg.friends.foreach { f =>
+          galaxyService ! GalaxyServiceMessage(GalaxyAction.LogStatusChange(f.name))
+        }
+      case _ => ()
+    }
+  }
+
+  def handleUseCooldownRenew(definition: BasicDefinition): Unit = {
+    definition match {
+      case _: KitDefinition => kitToBeUsed = None
+      case _ => ;
+    }
+  }
```

Comment:
for the method that are of kind:

```scala
def foo(a: A): B = {
    a match {
          case ...
          case ...
          case ...
   }
}
```

it is possible to change it to:

```scala
def foo: A => B = {
      case ...
      case ...
      case ...
}
```
and that does not change the way it is called `foo(a)` in both case.